third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/WSCodePartition.cpp

```diff
@@ -2205,6 +2205,52 @@ void insertAsyncComm(
                        headConsumer, consumerWaitPoint, isPost);
     }
   }
+
+  // Clean up tokens that are not used anymore.
+  // Remove an LocalAllocOp op if it is only used by
+  // MemDescIndexOp/InitBarrierOp
+  DenseSet<Value> removedBarriers;
+  auto removeTokenfNotUsed = [&](Value barrier) {
+    if (removedBarriers.count(barrier))
+      return;
+    if (barrier.use_empty()) {
+      barrier.getDefiningOp()->erase();
+      removedBarriers.insert(barrier);
+      return;
+    }
+
+    if (auto alloc = dyn_cast<ttg::LocalAllocOp>(barrier.getDefiningOp())) {
+      // Check: alloc result is only used once
+      if (!alloc->hasOneUse())
+        return;
```

Comment:
Is this the case where a LocalAllocOp allocates for multiple barriers? And you are not handling the case where all barriers of the LocalAllocOp are unused?

Reply:
Yes it's possible, but not with this parituclar kernel. Should be a TODO.